TypeDBConsole.java

```diff
@@ -343,10 +343,10 @@ private boolean runInlineCommandMode(CLIOptions options, List<String> inlineComm
         boolean[] cancelled = new boolean[]{false};
         terminal.handle(Terminal.Signal.INT, s -> cancelled[0] = true);
         try (TypeDBClient client = createTypeDBClient(options)) {
-            int i = 0;
-            for (; i < inlineCommands.size() && !cancelled[0]; i++) {
+            for (int i = 0; i < inlineCommands.size() && !cancelled[0]; i++) {
                 String commandString = inlineCommands.get(i);
                 printer.info("+ " + commandString);
+                if (commandString.startsWith("#")) continue;
```

Comment:
Note that `commandString` is a trimmed string, so if the source line has leading whitespace followed by `#`, this condition will still pass.